core/src/main/java/org/apache/accumulo/fate/zookeeper/ZooLock.java

```diff
@@ -139,26 +144,66 @@ public synchronized boolean tryLock(LockWatcher lw, byte[] data)
     return false;
   }
 
+  private static void sortChildrenByLockPrefix(List<String> children) {
+    Collections.sort(children, new Comparator<String>() {
+      @Override
+      public int compare(String o1, String o2) {
+
+        // Lock should be of the form:
+        // zlock#UUID#sequenceNumber
+        // Example:
+        // zlock#44755fbe-1c9e-40b3-8458-03abaf950d7e#0000000000
+
+        boolean lValid = true;
+        if (o1.length() <= 43) {
```

Comment:
Does a length of 43 or less imply there is no sequence number?

Reply:
Yes. FYI - I talked with Ed and he gave me some more insight into some of the issues involved. I'll be updating this PR based on his feedback.

Reply:
Okay sounds good, I'll keep an eye out for the updates. One thing that I'll add is the possibility of adding this functionality within ZooReaderWriter.java instead which would give us the flexibility to use the UUID with other kinds of nodes as well. I'm not sure if we would ever need this for other nodes besides the Ephemeral Sequential nodes used for ZooLock, but just thought I would point that out in case.